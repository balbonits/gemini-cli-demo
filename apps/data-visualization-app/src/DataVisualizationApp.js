class DataVisualizationApp extends HTMLElement {
    constructor() {
        super();
        this.attachShadow({ mode: 'open' });
        this.shadowRoot.innerHTML = `
            <style>
                .chart-container {
                    position: relative;
                    width: 100%;
                    max-width: 500px;
                    height: auto;
                    max-height: 400px;
                }
                .help-button {
                    position: absolute;
                    top: 10px;
                    right: 10px;
                    font-size: 1.5rem;
                    background: none;
                    border: none;
                    cursor: pointer;
                }
                .help-modal {
                    position: fixed;
                    top: 0;
                    left: 0;
                    width: 100%;
                    height: 100%;
                    background-color: rgba(0,0,0,0.5);
                    display: flex;
                    justify-content: center;
                    align-items: center;
                }
                .help-modal-content {
                    background-color: white;
                    padding: 2rem;
                    border-radius: 5px;
                    max-width: 500px;
                }
                .close-button {
                    position: absolute;
                    top: 10px;
                    right: 10px;
                    font-size: 1.5rem;
                    cursor: pointer;
                }
            </style>
            <div class="chart-container">
                <canvas></canvas>
            </div>
            <button class="help-button">?</button>
            <div class="help-modal" style="display: none;">
                <div class="help-modal-content">
                    <span class="close-button">&times;</span>
                    <h2>Data Visualization Help</h2>
                    <p>This is a simple data visualization app that displays a line chart.</p>
                    <p>The chart data is hardcoded in the component.</p>
                </div>
            </div>
        `;

        this.canvas = this.shadowRoot.querySelector('canvas');
        this.ctx = this.canvas.getContext('2d');

        this.helpButton = this.shadowRoot.querySelector('.help-button');
        this.helpModal = this.shadowRoot.querySelector('.help-modal');
        this.closeButton = this.shadowRoot.querySelector('.close-button');

        this.helpButton.addEventListener('click', () => {
            this.helpModal.style.display = 'flex';
        });

        this.closeButton.addEventListener('click', () => {
            this.helpModal.style.display = 'none';
        });

        this.renderChart();
    }

    renderChart() {
        const data = {
            labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
            datasets: [{
                label: 'Sample Data',
                data: [65, 59, 80, 81, 56, 55, 40],
                fill: false,
                borderColor: 'rgb(75, 192, 192)',
                tension: 0.1
            }]
        };

        new Chart(this.ctx, {
            type: 'line',
            data: data,
            options: {
                responsive: true
            }
        });
    }
}

customElements.define('data-visualization-app', DataVisualizationApp);