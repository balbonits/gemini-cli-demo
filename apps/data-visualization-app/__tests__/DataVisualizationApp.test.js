import '../src/DataVisualizationApp.js';

describe('DataVisualizationApp', () => {
  let dataVisualizationApp;

  beforeEach(() => {
    dataVisualizationApp = document.createElement('data-visualization-app');
    document.body.appendChild(dataVisualizationApp);
  });

  afterEach(() => {
    document.body.removeChild(dataVisualizationApp);
  });

  test('should render correctly', () => {
    expect(dataVisualizationApp).toMatchSnapshot();
  });
});