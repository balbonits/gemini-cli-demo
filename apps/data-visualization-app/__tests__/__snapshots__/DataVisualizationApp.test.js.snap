// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DataVisualizationApp should render correctly 1`] = `<data-visualization-app />`;
